query/sources/impl/src/test/groovy/org/codice/ddf/admin/sources/SourceTestCommons.groovy

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.sources
+
+import ddf.catalog.data.ContentType
+import ddf.catalog.operation.QueryRequest
+import ddf.catalog.operation.SourceResponse
+import ddf.catalog.service.ConfiguredService
+import ddf.catalog.source.Source
+import ddf.catalog.source.SourceMonitor
+import ddf.catalog.source.UnsupportedQueryException
+import org.codice.ddf.admin.common.fields.common.*
+import org.codice.ddf.admin.common.services.ServiceCommons
+import org.codice.ddf.admin.sources.fields.FactoryPid
+import org.codice.ddf.admin.sources.fields.ServicePid
+import org.codice.ddf.admin.sources.fields.type.SourceConfigUnionField
+
+class SourceTestCommons {
+
+    static final ENDPOINT_URL = SourceConfigUnionField.ENDPOINT_URL_FIELD
```

Comment:
I'm not going to review the testing yet, but just a heads up, you can create a new object field, populate it with the values you want, then call `getValue` on it. This will transform it into the map you need to pass as an arg to the actions.